packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts

```diff
@@ -26,15 +28,36 @@ const isNumber = (value: any): boolean => !isNaN(parseFloat(value)) && isFinite(
 export const isDateArray = (array: any[]): boolean => Array.isArray(array) && array.every(isDate);
 export const isNumberArray = (array: any[]): boolean => Array.isArray(array) && array.every(isNumber);
 
+const UseColorMapping = () => {
+  const [colorMapping, setColorMapping] = React.useState(new Map<string, string>());
+
+  const getColor = (colorString: string, index: number): { color: string; opacity?: number } => {
+    const d3ColorObj = d3Color(colorString);
+    const hexColor = d3ColorObj ? d3ColorObj.formatHex() : colorString;
+
+    if (!colorMapping.has(hexColor)) {
+      const nextColor = getNextColor(index);
+      setColorMapping(prev => new Map(prev).set(hexColor, nextColor));
+      return { color: nextColor, opacity: d3ColorObj?.opacity };
+    }
+
+    return { color: colorMapping.get(hexColor) as string, opacity: d3ColorObj?.opacity };
```

Comment:
don't use plotly opacity. It could make the color non accessible.

Reply:
done